src/fromager/commands/pypi_info.py

```diff
@@ -0,0 +1,155 @@
+    if requested_version:
+        logger.info(f"Requesting specific version: {requested_version}")
+
+    try:
+        package_info = _get_package_info(pypi_base_url, package_name, requested_version)
+        _display_package_info(package_info, package_name, requested_version)
+    except PackageNotFoundError as e:
+        logger.error(str(e))
+        raise click.ClickException(str(e)) from e
+    except Exception as e:
+        logger.error(f"Failed to retrieve package information: {e}")
+        raise click.ClickException(
+            f"Failed to retrieve package information: {e}"
+        ) from e
+
+
+class PackageNotFoundError(Exception):
+    """Raised when a package or version is not found on PyPI."""
+
+    pass
+
+
+def _get_package_info(
+    pypi_base_url: str, package_name: str, version: str | None = None
+) -> dict[str, Any]:
+    """Fetch package information from PyPI JSON API."""
+    if version:
+        url = f"{pypi_base_url}/{package_name}/{version}/json"
+    else:
+        url = f"{pypi_base_url}/{package_name}/json"
+
+    logger.debug(f"Requesting PyPI data from: {url}")
+
+    response = session.get(url)
+    if response.status_code == 404:
+        if version:
+            raise PackageNotFoundError(
+                f"Package '{package_name}' version '{version}' not found on PyPI"
+            )
+        else:
+            raise PackageNotFoundError(f"Package '{package_name}' not found on PyPI")
+
+    response.raise_for_status()
+    return response.json()
+
+
+def _display_package_info(
+    package_data: dict[str, Any], package_name: str, requested_version: str | None
+) -> None:
+    """Display package information in a structured format."""
+    info = package_data.get("info", {})
+    urls = package_data.get("urls", [])
+
+    # Package name and version
+    name = info.get("name", package_name)
+    version = info.get("version", "unknown")
+
+    print(f"Package: {name}")
+    print(f"Version: {version}")
+
+    # Package found status
+    print("Found on PyPI: Yes")
+
+    # License information
+    license_info = info.get("license") or "Not specified"
+    # Handle cases where license is empty string or None
+    if not license_info or license_info.strip() == "":
+        license_info = "Not specified"
+    print(f"License: {license_info}")
+
+    # URLs
+    home_page = info.get("home_page") or info.get("project_url")
+    if home_page:
+        print(f"Homepage: {home_page}")
+    else:
+        print("Homepage: Not specified")
+
+    # Check for other relevant URLs
+    project_urls = info.get("project_urls") or {}
+    if isinstance(project_urls, dict):
+        for url_type, url in project_urls.items():
+            if url_type.lower() in ("repository", "source", "github", "gitlab"):
+                print(f"Repository: {url}")
+                break
+
+    # Distribution types analysis
+    has_sdist = False
+    has_wheel = False
+
+    for url_info in urls:
```

Comment:
On recurring problem we have is packages with only prerelease versions, which are not resolved by default. It would be useful to show that sort of detail here when no version is specified.